packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,175 +70,261 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
+
+    script
+        .run(scope)
+        .ok_or_else(|| Error::Eval(path.into(), "execution error".to_string()))
 }
 
-fn import_file<'m>(
-    mv8: &'m MiniV8,
+fn import_file<'s>(
+    scope: &mut v8::HandleScope<'s>,
+    context: v8::Local<'s, v8::Context>,
     path: &str,
-    args: Vec<&mv8::Value<'m>>,
-) -> Result<mv8::Value<'m>> {
-    let v = eval_file(mv8, path)?;
-    let f = v
-        .as_function()
-        .ok_or_else(|| Error::FileImport(path.into(), format!("Failed to wrap file: {:?}", &v)))?;
-
-    let args = {
-        let mut a = Vec::new();
-        for arg in args {
-            a.push(arg.clone());
-        }
-        a
-    };
-    let args = mv8::Values::from_vec(args);
+    args: &[v8::Local<'s, v8::Value>],
+) -> Result<v8::Local<'s, v8::Value>> {
+    let v = eval_file(scope, path)?;
+    let f: v8::Local<'_, v8::Function> = v.try_into().map_err(|err| {
+        Error::FileImport(path.into(), format!("Failed to wrap file: {v:?}, {err}"))
+    })?;
+
+    let global_context = context.global(scope);
     let imported = f
-        .call(args)
-        .map_err(|e| Error::FileImport(path.into(), e.into()))?;
+        .call(scope, global_context.into(), args)
+        .ok_or_else(|| Error::FileImport(path.into(), "could not call function".to_string()))?;
 
     Ok(imported)
 }
 
-impl<'m> Embedded<'m> {
-    fn import(mv8: &'m MiniV8) -> Result<Self> {
-        let arrow = eval_file(mv8, "./src/worker/runner/javascript/apache-arrow-bundle.js")?;
-        let hash_stdlib = eval_file(mv8, "./src/worker/runner/javascript/hash_stdlib.js")?;
-        let hash_util = import_file(mv8, "./src/worker/runner/javascript/hash_util.js", vec![
-            &arrow,
-        ])?;
-        let batches_prototype = import_file(mv8, "./src/worker/runner/javascript/batch.js", vec![
-            &arrow, &hash_util,
-        ])?;
-
-        let ctx_import = import_file(mv8, "./src/worker/runner/javascript/context.js", vec![
-            &hash_util,
-        ])?;
-        let ctx_import = ctx_import.as_array().ok_or_else(|| {
+impl<'s> Embedded<'s> {
+    fn import(
+        scope: &mut v8::HandleScope<'s>,
+        context: v8::Local<'s, v8::Context>,
+    ) -> Result<Self> {
+        fn get_function_from_aray<'s>(
+            scope: &mut v8::HandleScope<'s>,
+            fns: v8::Local<'s, v8::Array>,
+            index: u32,
+        ) -> Result<v8::Local<'s, v8::Function>> {
+            fns.get_index(scope, index)
+                .ok_or_else(|| {
+                    Error::V8(format!("Could not get index {index} from runner.js array"))
+                })?
+                .try_into()
+                .map_err(|err| {
+                    Error::V8(format!(
+                        "Could not convert value at index {index} in runner.js as a function: \
+                         {err}"
+                    ))
+                })
+        }
+
+        let arrow = eval_file(
```

Comment:
I've made a task so we can see if we don't have to do this anymore
https://app.asana.com/0/1201893074552404/1202018802226446/f

While poking around the API did you see any ways to import files rather than doing the whole eval/run thing?